homeassistant/components/xiaomi_aqara.py

```diff
@@ -214,6 +215,7 @@ def __init__(self, device, name, xiaomi_hub):
         self._write_to_hub = xiaomi_hub.write_to_hub
         self._get_from_hub = xiaomi_hub.get_from_hub
         self._device_state_attributes = {}
+        self._unique_id = DOMAIN + "_" + slugify(self._name)
```

Comment:
No need to include DOMAIN.

I still think that `sid` should be fine. The name that you say it is concatenated with, where is it defined? And can anyone change it in upstream lib?

I think that we should just stick with serial . The odds of a serial number clashing are really really low?

Reply:
It's defined here:
https://github.com/tinloaf/home-assistant/blob/b6a83fafd12b1ad81801540f37ff53cd2b83b9e8/homeassistant/components/xiaomi_aqara.py#L214
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/binary_sensor/xiaomi_aqara.py#L227-L228
It's no part of the upstream library.

Reply:
There are zigbee devices (single sid/mac address) with multiple entity_ids: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/sensor/xiaomi_aqara.py#L22-L30

Reply:
Good catch @syssi - I did not think of the temperature sensors. :+1: 

@balloob Our platform implementation controls the names, thus there is no danger in them changing without our control. I will remove the domain.

Reply:
But could we get away with not using the name? I understand that there might be multiple entities from 1 device, but those could be distinguished by a type maybe?

Reply:
Well we could use introspection (i.e., `self.__class__.__name__`) to figure out what type of entity we are and use that instead of the name. Or did you mean something else by 'type'?

Reply:
We should rename the `name` variable to `type`.
I think that will make it more clear

Reply:
`self.__class__.__name__` wouldn't sufficient cp. https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/sensor/xiaomi_aqara.py#L22-L27

The data_key ("temperature") or name ("Temperature") is needed too for a unique entity.

Reply:
data_key, that's the one we should use for unique id.